CSharp/Samples/AadV1Bot/Dialogs/RootDialog.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System;
+using System.Threading.Tasks;
+using System.Configuration;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Connector;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System.Web;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using AdaptiveCards;
+using System.Linq;
+using System.Text;
+using System.Threading;
+
+namespace Microsoft.Bot.Sample.AadV1Bot.Dialogs
+{
+    [Serializable]
+    public class RootDialog : IDialog<object>
+    {
+        private static string ConnectionName = ConfigurationManager.AppSettings["ConnectionName"];
+
+        public async Task StartAsync(IDialogContext context)
+        {
+            context.Wait(MessageReceivedAsync);
+        }
+
+        private async Task MessageReceivedAsync(IDialogContext context, IAwaitable<object> result)
+        {
+            var activity = await result as Activity;
+            
+            var message = activity.Text.ToLowerInvariant();
+
+            if (message.ToLowerInvariant().Equals("recents"))
+            {
+                context.Call(CreateGetTokenDialog(), ListRecentMail);
+            }
+            else if (message.ToLowerInvariant().StartsWith("send"))
+            {
+                var recipient = message.ToLowerInvariant().Split(' ');
+                if (recipient.Length == 2)
+                {
+                    context.Call(CreateGetTokenDialog(), async (IDialogContext ctx, IAwaitable<GetTokenResponse> tokenResponse) => {
+                        await SendMail(context, tokenResponse, recipient[1]);
+                    });
+                }
+                else
+                {
+                    await context.PostAsync("You need to enter: 'send <recipient_email>' to send an email.");
+                }
+            }
+            else if (message.ToLowerInvariant().Equals("me"))
+            {
+                context.Call(CreateGetTokenDialog(), ListMe);
+            }
+            else if (message.ToLowerInvariant().Equals("signout"))
+            {
+                await Signout(context);
+            }
+            else
+            {
+                await context.PostAsync("You can type 'recents', 'send <recipient_email>', or 'me' to list things from AAD v1.");
+                context.Wait(MessageReceivedAsync);
+            }
+        }
+
+        public static async Task Signout(IDialogContext context)
+        {
+            await context.SignOutUserAsync(ConnectionName);
+            await context.PostAsync($"You have been signed out.");
+        }
+        
```

Comment:
As a sample I think you need to heavily comment this to explain how all of this works...